tests/model/cycle_data/test_perfherder_cycling.py

```diff
@@ -704,7 +704,7 @@ def test_one_month_worth_of_data_points(
         )
         perf_data.append(data)
 
-    timestamp += timedelta(days=30)
+    timestamp += timedelta(days=31)
```

Comment:
For my own understaning:
1. The failure had been observed on 2022-01-27.
2. The test runs with `nr_months=8` and the calculation of the timestamp for PerformanceDatums is `(nr_months - 1) * 30`. That's 2021-07-01.
3. Another PerformanceDatum gets added 30 days later: 2021-07-31.
4. Because both timestamps are in the same month, [this check](https://github.com/mozilla/treeherder/blob/cfb765d7b222b8e6a09e08187c0e98c4fb933135/treeherder/perf/models.py#L108-L123) returns `False` and {the signature is declared removable](https://github.com/mozilla/treeherder/blob/cfb765d7b222b8e6a09e08187c0e98c4fb933135/treeherder/model/data_cycling/removal_strategies.py#L347-L351).
This check means a data signature with records for the last day of the month and for the first day of the next month is declared relevant while first and last day of the same month are not. But that's not the scope of this change.